_posts/2021-06-25-app.md

```diff
@@ -0,0 +1,103 @@
+
+### Projektstruktur
+
+Um das Projekt möglichst simpel und aufgeräumt zu halten verwenden wir unterschiedlichste methodiken. Generell sind natürlich eine saubere Versionsverwaltung und der einsatz eines strengen Linters für alle unsere Applikationen pflicht. Darüber hinaus setzen wir bei der App auf eine moderne schreibweise von React unter der Verwendung von "Functional Components" und eine State Verwaltung unter Zuhilfenahme von React hooks.
+
+Da die Entwicklung in einem sehr beschränkten zeitlichen Rahmen stattfand haben wir an vielen Stellen auf gut gewartete Libraries zurückgegriffen.
+Besonders die Auswahl eines location providers unter Android ist uns dabeis chwer gefallen, da die native variante im Android core deutlich schwächere Erbenisse lieferte, als die durch Google Services bereitgestellte. Da wir uns entschieden haben ohne Google Services unsere App anbieten zu können hat sich nach einigen Tests ["react-native-geolocation"](https://github.com/react-native-geolocation/react-native-geolocation) als beste cross-platform Library herausgestellt.
+
+Um das Aussehen der App auf beiden Platfformen möglichst ähnlich zu halten und hier nicht für komplexere design elemente in die Tiefe der einzelnen Plattformen gehen zu müssen greifen wir auf [UI Kitten](https://akveo.github.io/react-native-ui-kitten/) zurück.
+
+### Mapbox Odysee
+
+Wie bereits erwähnt war für uns zu Beginn des Projekts klar, dass wir durch limitierte Ressourcen auf die Navigations Ansicht des Herstellers Mapbox zurückgreifen würden.
+
+Um unser Routing direkt in Mapbox Navigations nutzen zu können forkten wir die inoffizielle ReactNative Library und bauten sie zur Nutzung mit unserem eigenen Routing Backend um ([Link zum Fork](https://github.com/CargoRocket/react-native-mapbox-navigation)). Falls ein eigener Umbau angestrebt wird kann auch [dieses Issue](https://github.com/mapbox/mapbox-java/issues/1219) sehr hilfreich sein.
+
+Während der Entwicklung zeigte sich aber leider, dass das Pricing Schema das Mapbox für seine Navigationsansicht ansetzt sich nicht für unseren Usecase eignet.
+Durch Entwicklung alleine sind uns ungewollt dabei Kosten von 95$ entstanden. Dankenswerterweise konnten wir in Gesprächen mit Mapbox direkte Unterstützung erhalten, wodurch uns dieser Betrag final nicht in Rechnung gestellt wurde. Es war aber dennoch klar, dass eine weitere Nutzung von Mapbox Navigations für uns keine nachhaltige Option darstellt.
+
+Eine einfache alternative war gefragt, die lediglich Map-Tiles von Mapbox abruft. Die aktuelle Lösung für diese Ansicht besteht daher momentan aus einer eigenen Entwicklung auf Frameworkebene, unter Nutzung von UI Kitten Komponenten und ["@react-native-mapbox-gl/maps"](https://github.com/react-native-mapbox-gl/maps) zur Kartendarstellung.
+
+Die verwendete Datenstruktur orientiert sich dabei momentan stark an dem von Mapbox entwickelten Format. In Zukunft wollen wir hier eine direktere verwendung des Graphhopper Route Formats anstreben.
```

Comment:
Maybe you could extend this a bit, which "Datenstruktur" you mean exactly.
Also, you never mentioned GraphHopper before. Maybe it's usefule to introduce this, that the reader knows which data structure you convert it from.

Reply:
I've linked the formats. I really dont know what there could be more to say about it.

Reply:
If someone works themself threw this article I expect them to look the routing part up in our previous one.